pennylane/qaoa/cost.py

```diff
@@ -321,11 +323,16 @@ def max_independent_set(graph, constrained=True):
         raise ValueError("Input graph must be a nx.Graph, got {}".format(type(graph).__name__))
 
     if constrained:
-        return (bit_driver(graph.nodes, 1), qaoa.bit_flip_mixer(graph, 0))
+        cost_h = bit_driver(graph.nodes, 1)
+        cost_h.grouping_indices = [list(range(len(cost_h.ops)))]
+        return (cost_h, qaoa.bit_flip_mixer(graph, 0))
 
     cost_h = 3 * edge_driver(graph, ["10", "01", "00"]) + bit_driver(graph.nodes, 1)
```

Comment:
I suppose this is why we can't add the grouping indices to `bit_driver` and `edge_driver`, since the addition results in the loss of grouping info? Something to think about when we consider the operator overhaul :)